vowpalwabbit/csoaa.cc

```diff
@@ -34,22 +34,34 @@ namespace CSOAA
 struct csoaa
 {
   uint32_t num_classes = 0;
+  int indexing = -1;
+  bool search = false;
   polyprediction* pred = nullptr;
   ~csoaa() { free(pred); }
 };
 
 template <bool is_learn>
 inline void inner_loop(single_learner& base, example& ec, uint32_t i, float cost, uint32_t& prediction, float& score,
-    float& partial_prediction)
+    float& partial_prediction, int& indexing)
```

Comment:
```suggestion
    float& partial_prediction, int indexing)
```

if `indexing` isn't modified by the function it should be passed in by value since it's a built in type